src/service/navigation.js

```diff
@@ -535,63 +526,57 @@ export class Navigation {
   }
 
   /**
-   * Handles clicking on a link with hash navigation.
+   * Handles click-navigation on a non-A2A, standard-protocol link.
    * @param {!Event} e
-   * @param {!Element} target
-   * @param {!Location} tgtLoc
-   * @param {!Location} curLoc
+   * @param {!Element} element
+   * @param {!Location} targetLocation
+   * @param {!Location} currentLocation
    * @private
    */
-  handleNavClick_(e, target, tgtLoc, curLoc) {
-    const tgtHref = getHref(tgtLoc);
-    const curHref = getHref(curLoc);
-
-    // If the current target anchor link is the same origin + path
-    // as the current document then we know we are just linking to an
-    // identifier in the document. Otherwise, it's an external navigation.
-    if (!tgtLoc.hash || tgtHref != curHref) {
+  handleNavigation_(e, element, targetLocation, currentLocation) {
+    const target = getHrefMinusHash(targetLocation);
+    const current = getHrefMinusHash(currentLocation);
+
+    // Handle same-page (hash) navigation separately.
+    if (targetLocation.hash && target == current) {
+      this.handleHashNavigation_(e, targetLocation, currentLocation);
+    } else {
+      // Otherwise, this is an other-page (external) navigation.
       if (this.isEmbed_ || this.isInABox_) {
-        // Target in the embed must be either _top or _blank. If none specified,
-        // force to _blank.
-        const targetAttr = (target.getAttribute('target') || '').toLowerCase();
+        // Target in the embed must be either _top or _blank (default).
+        const targetAttr = (element.getAttribute('target') || '').toLowerCase();
         if (targetAttr != '_top' && targetAttr != '_blank') {
-          target.setAttribute('target', '_blank');
+          element.setAttribute('target', '_blank');
         }
-        return; // bail early.
       }
-
-      // Accessibility fix for IE browser.
-      // Issue: anchor navigation in IE changes visual focus of the browser
-      // and shifts to the element being linked to,
-      // where the input focus stays where it was.
-      // @see https://humanwhocodes.com/blog/2013/01/15/fixing-skip-to-content-links/
-      // @see https://github.com/ampproject/amphtml/issues/18671
-      if (Services.platformFor(this.ampdoc.win).isIe()) {
-        const internalTargetElmId = tgtLoc.hash.substring(1);
-        const internalElm = this.ampdoc.getElementById(internalTargetElmId);
-        if (internalElm) {
-          if (
-            !/^(?:a|select|input|button|textarea)$/i.test(internalElm.tagName)
-          ) {
-            internalElm.tabIndex = -1;
-          }
-          tryFocus(internalElm);
-        }
-      }
```

Comment:
AFAICT this was meant to handle hash navigation but it will never fire because the outer conditional specifically only passes on non-hash navigations.